packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder, PlaceholderAlignment;
+
+import 'package:flutter/painting.dart';
+
+import 'framework.dart';
+
+/// An immutable widget that is embedded inline within text.
+///
+/// The [child] property is the widget that will be embedded. It is the
+/// widget's responsibility to size itself appropriately, the text will
+/// not enforce any constraints.
+///
+/// The [child] property may contain its own [Widget] children (if applicable),
+/// including [Text] and [RichText] widgets which may include additional
+/// [WidgetSpan]s. Child [Text] and [RichText] widgets will be laid out
+/// independently and occupy a rectangular space in the parent text layout.
+///
+/// [WidgetSpan]s will be ignored when passed into a [TextPainter] directly.
+/// To properly layout and paint the [widget], [WidgetSpan] should be passed
+/// into a [Text.rich] widget.
+///
+/// {@tool sample}
+///
+/// A card with `Hello World!` embedded inline within a TextSpan tree.
+///
+/// ```dart
+/// <TextSpan>[
+///   TextSpan(text: 'Flutter is'),
+///   WidgetSpan(
+///     widget: SizedBox(
+///       width: 120,
+///       height: 50,
+///       child: Card(
+///         child: Center(
+///           child: Text('Hello World!')
+///         )
+///       ),
+///     )
+///   ),
+///   TextSpan(text: 'the best!'),
+/// ]
+/// ```
+/// {@end-tool}
+///
+/// The [semanticsLabel] argument may be provided to supply a description of
+/// this [WidgetSpan] within the [toPlainText]. The semantics label of the
+/// overall [InlineSpan] tree will not mention [WidgetSpan]s unless
+/// [semanticsLabel] is provided. Semantics for the widget itself will still
+/// function independently from the [InlineSpan] semantics label.
+///
+/// See also:
+///
+///  * [TextSpan], a node that represents text in a [InlineSpan] tree.
+///  * [Text], a widget for showing uniformly-styled text.
+///  * [RichText], a widget for finer control of text rendering.
+///  * [TextPainter], a class for painting [InlineSpan] objects on a [Canvas].
+@immutable
+class WidgetSpan extends PlaceholderSpan {
+  /// Creates a [WidgetSpan] with the given values.
+  ///
+  /// The [child] property should be non-null. [WidgetSpan] is a leaf node in
+  /// the [InlineSpan] tree. Child widgets are constrained by the width of the
```

Comment:
In the class-level comment it was said that no constraints are enforced. This sounds like there are constraints that get enforced?